core/lib/types/src/utils.rs

```diff
@@ -77,6 +77,38 @@ pub fn deployed_address_create(sender: Address, deploy_nonce: U256) -> Address {
     Address::from_slice(&keccak256(&bytes)[12..])
 }
 
+/// Packs all pubdata needed for batch commitment in boojum into one bytes array. The packing contains the
+/// following: logs, messages, bytecodes, and compressed state diffs.
+/// This data is currently part of calldata but will be submitted as part of the blob section post EIP-4844.
+pub fn construct_pubdata(l1_batch_with_metadata: &L1BatchWithMetadata) -> Vec<u8> {
```

Comment:
This was a method on the structure, but now it's a free function that still accepts the same argument (e.g. it was `&self` before, now it's `l1_batch_with_metadata`). I think it slightly worsens the logical structure of the crate (e.g. it was in an expected namespace -- as a method of the corresponding function -- but now it's still in the global namespace, but in an unexpected place).

Is there any strong reasoning for it? If not, I'd suggest to roll it back.

Reply:
In terms of design, there are two things: one is that it's a little bit smelly that this was a method for `L1BatchWithMetadata` as it doesn't make a lot of sense for such an entity to be responsible for pubdata construction. Secondly, the responsibility of whether the pubdata is constructed or not was delegated to the new abstraction and we kinda can say again that it doesn't make sense anymore where it is. What do you think?

If this thought is not enough for making such a change, we roll back and let the method where it was and then the abstraction, as it knows the `L1BatchWithMetadata`, could call it without any further issues.

Reply:
> In terms of design, there are two things: one is that it's a little bit smelly that this was a method for L1BatchWithMetadata as it doesn't make a lot of sense for such an entity to be responsible for pubdata construction. 

While I certainly agree with this statement, I'm still not convinced that a free function in a generally random module would express the domain adherence better. If we are to consider separation of interfaces and/or domain knowledge, I think it has to do with better encapsulation in the first place, and `utils` module probably doesn't provide any improvements on that front.

> Secondly, the responsibility of whether the pubdata is constructed or not was delegated to the new abstraction and we kinda can say again that it doesn't make sense anymore where it is. What do you think?

Given the above, I think that _majorly_ the design hasn't changed, the overall logic is still the same: it's either to call this method or not to call this method.

---

So, to summarize, I feel like this function was moved from one inappropriate place to another inappropriate place. I'd suggest to roll it back for now. But to your point, it certainly makes sense to revisit this place better to find a better logical structure.

Reply:
Tracked in https://github.com/lambdaclass/zksync-era/issues/141